src/source/raster_tile_source.js

```diff
@@ -221,10 +222,9 @@ class RasterTileSource extends Evented implements Source {
         tile.setTexture(data, painter);
     }
 
+    // eslint-disable-next-line no-unused-vars
     static unloadTileData(tile: Tile, painter: Painter) {
-        if (tile.texture) {
-            painter.saveTileTexture(tile.texture);
-        }
+        // Texture caching on unload occurs in unloadTile
```

Comment:
@stepankuzmin can you please check the logic here re `CustomSource`? Looks like the way this is currently wired, after the change, it would stop caching textures — what's the best way to accommodate the new approach while making sure nothing regresses there?

Reply:
Hey all! I've checked the CustomSource caching, and it was affected by this PR. I've made a fix for that in https://github.com/mapbox/mapbox-gl-js/pull/12993. Since this PR was made from a fork, I couldn't target my changes on top of it. We can merge them sequentially. /cc @mourner